env0/resource_custom_flow_assignment.go

```diff
@@ -0,0 +1,119 @@
+package env0
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/env0/terraform-provider-env0/client"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func setCustomFlowAssignmentId(d *schema.ResourceData, assignment client.CustomFlowAssignment) {
+	d.SetId(fmt.Sprintf("%s|%s", assignment.ScopeId, assignment.BlueprintId))
+}
+
+func getCustomFlowAssignmentFromId(d *schema.ResourceData) client.CustomFlowAssignment {
+	id := d.Id()
+	split := strings.Split(id, "|")
+
+	var assignment client.CustomFlowAssignment
+	assignment.ScopeId = split[0]
+	if len(split) > 1 {
+		assignment.BlueprintId = split[1]
+	}
+
+	assignment.Scope = client.CustomFlowProjectScope
+
+	return assignment
+}
+
+func resourceCustomFlowAssignment() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceCustomFlowAssignmentCreate,
+		ReadContext:   resourceCustomFlowAssignmentRead,
+		DeleteContext: resourceCustomFlowAssignmentDelete,
+
+		Schema: map[string]*schema.Schema{
+			"scope": {
+				Type:        schema.TypeString,
+				Description: "the type of the scope. Valid values: PROJECT",
+				Optional:    true,
+				Default:     client.CustomFlowProjectScope,
```

Comment:
For now, the only valid scope is "PROJECT". This simplifies the implementation.

Reply:
Please add this as a comment in the code
Or consider have a var `allowedCustomFlowAssignmentScope = client.CustomFlowProjectScope` and use that instead of using the scope in several places.

Reply:
comments added. Please let me know if it suffices.

Reply:
4355aab0d55f3a546b7880c7841d3f39981f6002